### ISSUE TITLE
Use traceroute to determine if an AutoNAT server is acceptable

### ISSUE THREAD

**Issue by Stebalien** [libp2p/go-libp2p-autonat#46]
We should consider using some form of traceroute on IPv6 to determine if a potential AutoNAT server is viable. This should be very fast because we just want to verify that we hit _some_ public IP on the route.

If we can get this working correctly, we can skip the other heuristics (although some of them may be a good initial check).

**Comment by Stebalien** [comment]
Yeah, it looks like we may be able to do a UDP simple UDP traceroute. We're not trying to get all the way to the target, just out of our network.

**Comment by willscott** [comment]
They'll all be "public" IPs, so we'll need some heuristic for deciding that we've hit some router along the way with an IP that diverges significantly from those of the endpoints.

Could we get away in many cases with just latency as an estimator of being in the same network?

**Comment by Stebalien** [comment]
Ah, because IPv6... Actually, we there's no reason not to try IPv4, right (if the peer has an IPv4 address)?

**Comment by Stebalien** [comment]
> Could we get away in many cases with just latency as an estimator of being in the same network?

Maybe? But I'd want to set _that_ limit to be _really_ short (like 1-5ms).
